framework/test/unit/specs/modules/dpos/transaction_assets/vote_transaction_asset.spec.ts

```diff
@@ -0,0 +1,812 @@
+				);
+			});
+
+			it('should make downvoted delegate account to have correct totalVotesReceived', async () => {
+				await transactionAsset.applyAsset(applyInput);
+
+				const updatedDelegate1 = await stateStoreMock.account.get<DPOSAccountProps>(
+					delegate1.address,
+				);
+				const updatedDelegate2 = await stateStoreMock.account.get<DPOSAccountProps>(
+					delegate2.address,
+				);
+
+				expect(updatedDelegate1.dpos.delegate.totalVotesReceived).toEqual(BigInt(0));
+				expect(updatedDelegate2.dpos.delegate.totalVotesReceived).toEqual(BigInt(0));
+			});
+		});
+
+		describe('when asset.votes contain negative and positive amount ', () => {
+			const positiveVoteDelegate1 = liskToBeddows(10);
+			const negativeVoteDelegate2 = liskToBeddows(-20);
+
+			beforeEach(async () => {
+				const votesInput = objects.cloneDeep(applyInput);
+				votesInput.asset = {
+					votes: [
+						{ delegateAddress: delegate1.address, amount: delegate1VoteAmount },
+						{ delegateAddress: delegate2.address, amount: delegate2VoteAmount },
+					],
+				};
+				await transactionAsset.applyAsset(votesInput);
+
+				// Clears mock calls for earlier apply asset
+				(applyInput.reducerHandler.invoke as jest.Mock).mockClear();
+
+				applyInput.asset = {
+					votes: [
+						{ delegateAddress: delegate1.address, amount: positiveVoteDelegate1 },
+						{ delegateAddress: delegate2.address, amount: negativeVoteDelegate2 },
+					].sort((a, b) => -1 * a.delegateAddress.compare(b.delegateAddress)),
+				};
+			});
+
+			it('should not throw error', async () => {
+				await expect(transactionAsset.applyAsset(applyInput)).resolves.toBeUndefined();
+			});
+
+			it('should make account to have correct balance', async () => {
+				await transactionAsset.applyAsset(applyInput);
+
+				expect(applyInput.reducerHandler.invoke).toHaveBeenCalledTimes(1);
+				expect(applyInput.reducerHandler.invoke).toHaveBeenCalledWith('token:debit', {
+					address: applyInput.senderID,
+					amount: positiveVoteDelegate1,
+				});
+			});
+
+			it('should make account to have correct unlocking', async () => {
+				await transactionAsset.applyAsset(applyInput);
+
+				const updatedSender = await stateStoreMock.account.get<DPOSAccountProps>(sender.address);
+				expect(updatedSender.dpos.unlocking).toHaveLength(1);
+				expect(updatedSender.dpos.unlocking).toEqual([
+					{
+						delegateAddress: delegate2.address,
+						amount: BigInt(-1) * negativeVoteDelegate2,
+						unvoteHeight: lastBlockHeight + 1,
+					},
+				]);
+			});
+
+			it('should make upvoted delegate account to have correct totalVotesReceived', async () => {
+				await transactionAsset.applyAsset(applyInput);
+
+				const updatedDelegate1 = await stateStoreMock.account.get<DPOSAccountProps>(
+					delegate1.address,
+				);
+
+				expect(updatedDelegate1.dpos.delegate.totalVotesReceived).toEqual(
+					delegate1VoteAmount + positiveVoteDelegate1,
+				);
+			});
```

Comment:
order of votes and unlocking should be checked?

Reply:
For checking the order there are separate test cases both for `sentVotes` and `unlocking`